### ISSUE TITLE
Text selectable does not allow normal cursor text selection

### ISSUE THREAD

**Issue by ericvicenti** [microsoft/react-native-macos#906]
### Environment

```markdown
react-native-macos@0.64.10
```

### Steps to reproduce the bug

```jsx
  <Text selectable={true}>
    "Selectable" text. Good luck selecting part of this string!
  </Text>
```

### Expected Behavior

Expect the cursor to be able to click and drag to select part of the Text children

### Actual Behavior

<img width="766" alt="Screen Shot 2021-11-18 at 11 27 18 AM" src="https://user-images.githubusercontent.com/1483597/142483835-7cf635c6-9d78-4a89-8c47-58cb37db901e.png">

**Comment by HeyImChris** [comment]
Thanks for filing- we're able to repro this as well. It looks like the macOS control is taking on some of the iOS behavior instead of being catered to desktop. Possible suspect is in [rightMouseDown](https://github.com/microsoft/react-native-macos/blob/f88f01a99e0f6fdd33b4e5c6e8ab882aba07b9ef/Libraries/Text/Text/RCTTextView.m#L375) in `RCTTextView`.

**Comment by adriandiaz** [comment]
Are there any plans for this to get fixed?

**Comment by Saadnajmi** [comment]
This is fixed by #1346 and will be included in our next major release (currently planned to be 0.70)
